pkgs/development/libraries/science/math/cudnn/generic.nix

```diff
@@ -17,44 +23,53 @@
 , url
 , hash ? null
 , sha256 ? null
-, supportedCudaVersions ? []
+, supportedCudaVersions ? [ ]
 }:
 
 assert (hash != null) || (sha256 != null);
 
+assert useCudatoolkitRunfile || (libcublas != null);
+
 let
+  inherit (cudatoolkit) cc;
+
   majorMinorPatch = version: lib.concatStringsSep "." (lib.take 3 (lib.splitVersion version));
   version = majorMinorPatch fullVersion;
-in stdenv.mkDerivation {
-  name = "cudatoolkit-${cudatoolkit.majorVersion}-cudnn-${version}";
 
+  cudatoolkit_root =
+    if useCudatoolkitRunfile
+    then lib.warn "Building cudnn against run-file based cudatoolkit instead of redist packages" cudatoolkit
```

Comment:
[OfBorg doesn't particularly like this](https://gist.github.com/GrahamcOfBorg/bbf9fba8fe9b8821d2a5e6f321ff0b6c)

Reply:
Deleted `lib.warn`